students/ScottL/session09/cmi_main.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python3
+
+"""
+Purpose: provide user interaction with Donor data
+Developed by: Scott Lipp
+Revision History
+    3/16/19 -- Created
+"""
+
+
+import sys
+import mailroom_donor as md
+
+
+class CommLineInter:
+    """For enabling the user to interface with Donor data."""
+
+    MENU_HEADER_1 = "\nMenu Options:"
+    MENU_HEADER_2 = f"\n{'-' * len(MENU_HEADER_1)}"
+    MENU_SELECTIONS = {1: "Add new donation and send 'Thank You' note",
+                       2: "Display a report",
+                       3: "Send 'Thank You' letters to all donors",
+                       4: "Quit"}
+    REPORT_HEADER_1 = "\n{:<26}{:<12}{:^17}{:>11}".format("Donor Name", "Total Dollars",
+                                                          "Number", "Avg Amount")
+    REPORT_HEADER_2 = f"\n{'-' * len(REPORT_HEADER_1)}"
+    DONOR_LIST_HEADER = f"\nDonor Name:\n{'-' * 20}"
+
+    def display_user_menu(self):
+        """Display the contents of the menu options for the user."""
+        print(self.MENU_HEADER_1, self.MENU_HEADER_2)
+        for item in self.MENU_SELECTIONS:
+            print(f"{item}: {self.MENU_SELECTIONS[item]}")
+
+    @staticmethod
+    def get_donor_name(first_last):
+        """
+        :param first_last: string to indicate which name (first or last) to supply
+        :return: string response to prompt
+        """
+        prompt = f"Please enter the {first_last} name of the donor, " \
+                 "'List' to see a list of existing donors, " \
+                 "or 'Quit' to return to the Main Menu: "
+        return input(prompt)
+
+    def display_donor_list(self):
+        """Display all the stored donor names for the user."""
+        if md.DonorCollection().name_list():
+            print(self.DONOR_LIST_HEADER)
+            for name in md.DonorCollection().name_list():
+                print(name)
+        else:
+            print("\n\tThe name list is empty.")
+
+    def validate_name(self):
+        """
+        Process the expected user input
+        :return: 2-element tuple containing the first and last names
+        """
+        first = self.get_donor_name("FIRST")
```

Comment:
you could have gotten the full name of the donor, and then run your logic for `quit` and `list` and then else return the name.
```
name = self.get_donor_name("first and last").lower()   # you could just ask for name instead of first/last in your        text

if name == "list":
    self.display_donor_list()
elif name == "quit":
    return
else:
    return name
```